lib/collector_registration.rb

```diff
@@ -0,0 +1,139 @@
+    else
+      exit(1)
+    end
+  end
+
+  def verify_connection_retrieving_missing_info
+    retrieve_organization_name
+    if !@configuration.blank_value?(:uc6_organization_name)
+      @configuration[:verified_api_connection] = true
+      logger.info "Succesful integration with uc6 api"
+    end
+  end
+
+  def uc6_api_configured?
+    !@configuration.blank_value?(:uc6_api_host) && is_valid_user?
+  end
+
+  def is_valid_user?
+    @configuration.blank_value?(:uc6_refresh_token) ?
+      validate_email & validate_password :
+      true
+  end
+
+  def validate_email
+    if ( !@configuration.blank_value?(:uc6_login_email) and
+    	@configuration[:uc6_login_email].match(/\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i) )
+      true
+    else
+  	  false
+    end
+  end
+
+  def validate_password
+    !@configuration.blank_value?(:uc6_login_password)
+  end
+
+  def oauth_succesful?
+    hyper_client = HyperClient.new
+    begin
+      # On first registration, these will already be blank. But if a user wanted to change a pasword,
+      #  we need to force a token request, so we make sure these are blanked out
+      GlobalConfiguration::GlobalConfig.instance.delete(:uc6_oauth_token)
+      hyper_client.reset_token
+      # trigger retrieval of new token
+      if ( hyper_client.oauth_token.blank? )
+        puts 'access token could not be retrieved. Please verify UC6 API options.'
+        return false
+       else
+       	return true
+      end
+    raise
+    rescue StandardError => e
+      if ( can_connect?(:uc6_api_host, URI.parse(@configuration[:uc6_api_host]).port) )
+        if ( e.is_a?(OAuth2::Error) )
+          # The OAuth2 error messages are useless, so we don't bother showing it to the user
+          logger.error "Authentication token could not be retrieved. Please verify the UC6 API credentials."
+        else
+          logger.error "Authentication token could not be retrieved: #{e.message}"
+        end
+      #else Errors are added as part of can_connect? method
+      end
+    end
+  end
+
+  def can_connect?(field, port)
+    begin
+      Timeout::timeout(10) {
+        host = @configuration[field].slice(%r{(?:https*://)?([^:]+)(?::\d+)*}i, 1)
+        Resolv.new.getaddress(host)
+        TCPSocket.new(host, port).close
+        true }
+    rescue Timeout::Error => e
+      logger.error "#{field} : timed during check: #{e.message}"
+      false
+    rescue Resolv::ResolvError => e
+      logger.error "#{field} :could not be resolved: #{e.message}"
+      false
+    rescue Errno::ECONNREFUSED => e
+      logger.error "#{field} :could not be connected to on port #{port}: #{e.message}"
+      false
+    rescue StandardError => e
+      logger.error "#{field} :could not be validated: #{e.message}"
+      false
+    end
+  end
+
+  def retrieve_organization_name
+    hyper_client = HyperClient.new
+    url = "#{@configuration[:uc6_api_host]}/api/v2/organizations/#{@configuration[:uc6_organization_id]}.json"
```

Comment:
Please verify that we are using all configuration details, as this time the URL will be "hardcoded" to a V2 of an api.